mantidproject/mantid
# Dev2017 flag and skip slow tests

A function `excludeInPullRequests` has been added to the base class for system tests, allowing for especially slow or unnecessary tests to be skipped for every PR build and only run nightly. Tests will be excluded if this function returns true.

When calling `systemtest` from the command line, all tests will be run by default. A flag `--exclude-in-pull-requests` has been added to exclude any systems test with the above function returning true.

A few of the slowest system tests have been updated to be skipped, see https://docs.google.com/spreadsheets/d/1uzAhnFM9Q8jMmrOzaZ32oBx_SDIdojKUgps9-EXpslo/edit#gid=0

**To test:**

Run a system test of your choice. 
Add the function
```
def excludeInPullRequests(self):
    return True
```
to the class of your system test.

run `systemtest -R yourtest` and it should be run.

run `systemtest -R yourtest --exclude-in-pull-requests` and it should be skipped.

Fixes #19864

**Release Notes** 


---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
